# (Auto)Trinotate notes for self

## AutoTrinotate script

AutoTrinotate script's help prints the following message to the terminal:

```
##############################################################################
#
# Required:
#
#  --Trinotate_sqlite <string>                Trinotate.sqlite boilerplate database
#
#  --transcripts <string>                     transcripts.fasta
#
#  --gene_to_trans_map <string>               gene-to-transcript mapping file
#
#  --conf <string>                            config file
#
#  --CPU <int>                                number of threads to use.
#
#
##############################################################################
```

Let's go over the required parameters:

* **Trinotate_sqlite:** An sqlite database generated by Trinotate's [Build_Trinotate_Boilerplate_SQLite_db.pl](https://github.com/Trinotate/Trinotate/blob/master/admin/Build_Trinotate_Boilerplate_SQLite_db.pl). Obs: This scripts uses several URLs to download data. Such pages might be moved or even deleted. In this case, you need to find where that specific resource is being hosted and add its URL to the script.

```
## Resources:
my $SPROT_DAT_URL = "ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.dat.gz";
my $EGGNOG_DAT_URL = "http://eggnogdb.embl.de/download/latest/data/NOG/NOG.annotations.tsv.gz";
my $GENE_ONTOLOGY_DAT_URL = "http://purl.obolibrary.org/obo/go/go-basic.obo";
my $PFAM_DAT_URL = "ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz";
#my $PFAM2GO_DAT_URL = "http://www.geneontology.org/external2go/pfam2go"; # ftp://ftp.geneontology.org/pub/go/external2go/pfam2go
my $PFAM2GO_DAT_URL = "ftp://ftp.geneontology.org/pub/go/external2go/pfam2go";
```


* **transcripts:** The result of a previous Trinity assembly
* **gene_to_trans_map:** A file linking geneids to transcripts ids. More on that [here](https://github.com/trinityrnaseq/trinityrnaseq/wiki/Output-of-Trinity-Assembly). To obtain this file, you have to run Trinity's [get_Trinity_gene_to_trans_map.pl](https://github.com/trinityrnaseq/trinityrnaseq/blob/master/util/support_scripts/get_Trinity_gene_to_trans_map.pl). Note that this script is in your **Trinity** installation directory, not in Trinotate's. To run it:

```
/path/to/trinity_dir/util/support_scripts/get_Trinity_gene_to_trans_map.pl Trinity_assembly.fa > Trinity_assembly_gene_to_trans
```


* **conf:** Path to [configuration file](https://github.com/Trinotate/Trinotate/blob/master/auto/conf.txt). This is where you will add the path to the programs and databases used by Trinotate. This is also the file that needs to be modified in order to [get your custom databases working in autoTrinotate](#running-autotrinotate-with-a-custom-swissprot-database)

```
[GLOBALS]

#  ** edit the progs and dbs section to point to your local resources.

# progs
TRANSDECODER_DIR=/home/unix/bhaas/GITHUB/TransDecoder
BLASTX_PROG=blastx
BLASTP_PROG=blastp
SIGNALP_PROG=/seq/regev_genome_portal/SOFTWARE/SIGNALP/signalp-4.0/signalp
TMHMM_PROG=/seq/regev_genome_portal/SOFTWARE/TMHMM/TMHMM2.0c/bin/tmhmm
RNAMMER_TRANS_PROG=/home/unix/bhaas/GITHUB/Trinotate/util/rnammer_support/RnammerTranscriptome.pl
RNAMMER=/seq/regev_genome_portal/SOFTWARE/RNAMMER/rnammer
HMMSCAN_PROG=hmmscan

# dbs
SWISSPROT_PEP=/seq/RNASEQ/DBs/TRINOTATE_RESOURCES/TRINOTATE_V3/uniprot_sprot.pep
PFAM_DB=/seq/RNASEQ/DBs/TRINOTATE_RESOURCES/TRINOTATE_V3/Pfam-A.hmm
```

* **CPU:** Number of threads. Pretty self-explanatory

## Running AutoTrinotate with a custom Swissprot database:

If you need Trinotate's BlastX adn BlastP analyses to be run against a custom Swissprot database (not sure if it works for TrEMBL), you need to perform the following steps:

1. Select the desired subset of data from the bank and download it as ".txt" (which is equivalent to the ".dat" format)
2. Run the [EMBL_swissprot_parser.pl](https://github.com/Trinotate/Trinotate/blob/master/admin/util/EMBL_swissprot_parser.pl):

```
/Path/to/Trinotate-directory/admin/util/EMBL_swissprot_parser.pl custom_data_sprot.txt prefix_name
```

3. This will generate the file "custom_data_sprot.txt.pep", used to create a blast database with makeblastdb:

```
makeblastdb -in custom_data_sprot.txt.pep -dbtype prot -out custom_data_sprot.txt.pep
```

4. Finally, update the ['conf.txt' file](https://github.com/Trinotate/Trinotate/blob/master/auto/conf.txt), adding the path to your custom SwissProt database.

```
# dbs
SWISSPROT_PEP=/path/to/custom_data_sprot.txt.pep
```

## Generating a summary of Trinotate results



## Rnammer

Probably the most inglorious task when setting Trinotate up is to install Rnammer. You can find some help at [Trinotate's wiki](https://github.com/Trinotate/Trinotate.github.io/wiki/Software-installation-and-data-required#rnammer-free-academic-download) and [here](https://github.com/gavieira/rnammer_trinotate).
